src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -274,7 +276,27 @@ private void requestBlockToBuildingIfRequired(final BuildingBuilder building, fi
             }
         }
 
-        building.addNeededResource(BlockUtils.getItemStackFromBlockState(blockState), 1);
+        if (!ChiselAndBitsCheck.isChiselAndBitsBlock(blockState))
+        {
+            building.addNeededResource(BlockUtils.getItemStackFromBlockState(blockState), 1);
+        }
+    }
+
+    @Override
+    public int getTotalRequiredAmount(final ItemStack deliveredItemStack)
+    {
+        if(getOwnBuilding() instanceof BuildingBuilder)
+        {
+            if (ItemStackUtils.isEmpty(deliveredItemStack))
+            {
+                return 0;
+            }
+            final int hashCode = deliveredItemStack.hasTagCompound() ? deliveredItemStack.getTagCompound().hashCode() : 0;
+            final BuildingBuilderResource resource
+                    = ((BuildingBuilder) getOwnBuilding()).getNeededResources().get(deliveredItemStack.getUnlocalizedName() + ":" + deliveredItemStack.getItemDamage() + "-" + hashCode);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 185 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)